src/builder/value_parser.rs

```diff
@@ -1906,6 +1906,20 @@ impl ValueParserFactory for u64 {
     }
 }
 
+impl ValueParserFactory for isize {
+    type Parser = RangedU64ValueParser<isize>;
+    fn value_parser() -> Self::Parser {
+        RangedU64ValueParser::new()
+    }
+}
+
+impl ValueParserFactory for usize {
+    type Parser = RangedU64ValueParser<usize>;
+    fn value_parser() -> Self::Parser {
+        RangedU64ValueParser::new()
+    }
+}
```

Comment:
I'm still torn on whether it is ok-enough to reuse i64/u64 or if we should make distinct usize/usize parsers.  I lean towards keeping using distinct parsers for now.

In the ideal case, we'd have `RangedValueParser<P=u64, T=u64>` which would allow choosing the parsed-type `P` (without pulling in the num traits crate).  I had some problems last time I looked into this.  Fully switching to it will require a breaking change but it shouldn't be too disruptive.

Reply:
What if there was a more general range type that looked something like:


```rust

pub struct RangedValueParser<T:  std::str::FromStr> {
  lower: std::ops::Bound<T>,
  upper: std::ops::Bound<T>,
}

impl<T> RangedValueParser where T: std::str::FromStr + std::cmp::PartialOrd + Clone {

  pub fn new() -> Self {
    Self::from(..)
  }

  pub fn range<B: RangeBounds<T>>(mut self, range: B) -> Self {
    unimplemented!("TODO");
  }
}

impl<T: std::str::FromStr + std::cmp::PartialOrd + Clone, B: RangeBounds<T>> From<B> for RangedValueParser<T> {
  fn from(range: B) -> Self {
    Self {
      lower: range.start_bound().cloned(),
      upper: range.end_bound().cloned(),
    }
  }
}

// then implement `RangedValueParser` for integer types, f32, and f64
```
One nice thing about that is it could be used for floating point numbers (for example specify a floating point number between 0 and 1), 128 bit integers (u128 and i128), or even arbitrary precision numbers from a library (although due to the orphan rule, would probably require a newtype). Although, for backwards compatibility it would probably need to be an alternative to `RangedU64ValueParser` rather than a full replacement, at least until 4.0

Reply:
I ran into problems when trying to implement it but if you get it working, that'd be great!